### DISCUSSION TITLE
setBackgroundMessageHandler only from background mode, not from quit

### DISCUSSION THREAD

**Discussion by assaf-malki** [invertase/react-native-firebase#5035] [Q&A]
Hey,
Is it possible for setBackgroundMessageHandler to be launched only when the app is in background state and not from quit state?
I have recently updated react-native-firebase package to v11, and I noticed there is some delay for setBackgroundMessageHandler to launch when in quit mode.
This can cause some race condition in my app between the headless task and the app being launched when the user press the notification message (on android).
If not possible, is there some indication that setBackgroundMessageHandler launched from background state and not from quit state?
Thanks

**Comment by mikehardy** [comment]
I think you can query the AppState react-native information and behave accordingly in your app logic, yes? https://reactnative.dev/docs/appstate#currentstate
